src/System.Diagnostics.EventLog/src/System/Diagnostics/EventLogEntry.cs

```diff
@@ -0,0 +1,506 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Text;
+using Microsoft.Win32;
+
+namespace System.Diagnostics
+{
+    [
+    ToolboxItem(false),
+    DesignTimeVisible(false),
+    Serializable,
+    ]
+    public sealed class EventLogEntry : Component, ISerializable
+    {
+        internal byte[] dataBuf;
+        internal int bufOffset;
+        private EventLogInternal owner;
+        private string category;
+        private string message;
+
+        internal EventLogEntry(byte[] buf, int offset, EventLogInternal log)
+        {
+            this.dataBuf = buf;
+            this.bufOffset = offset;
+            this.owner = log;
+
+            GC.SuppressFinalize(this);
+        }
+
+        private EventLogEntry(SerializationInfo info, StreamingContext context)
```

Comment:
The ctor here needs to throw too, e.g. like done here: https://github.com/dotnet/corefx/pull/20220/files#diff-2fd793fe0c269c716048c5b7c45b6e5bR24

Reply:
Do I need to make a new constructor and if not at which line I need to add the throw exception line

Reply:
Just remove the content of the private constructor and add the throw in there